src/charm.py

```diff
@@ -59,6 +63,21 @@ def __init__(self, *args: typing.Any) -> None:
         )
         self._observability = Observability(charm=self, charm_state=self._charm_state)
 
+    def _precondition_check(self) -> bool:
+        """Check preconditions for charm execution are met.
+
+        Returns: True if the precondition is met, False otherwise.
+        """
+        if self._sentinel.is_sentinel_image():
+            self._update_app_and_unit_status(
+                BlockedStatus(
+                    "charm requires a Flask app image, "
+                    "redeploy with '--resource flask-app-image=<your-image>'"
```

Comment:
I think you can also use `attach-resource` to do this without having to redeploy. I realise we have limited space here in terms of the message we display, but giving people a way to get out of this without needing to redeploy seems like it might be worthwhile.